src/model/test/Space_GTest.cpp

```diff
@@ -2013,6 +2027,90 @@ TEST_F(ModelFixture, Space_intersectSurfaces_degenerate3) {
   //EXPECT_NEAR(interiorFloorArea, 412.9019, 0.01);
   EXPECT_NEAR(exteriorRoofArea, 825.8048, 0.01);
   //EXPECT_NEAR(interiorRoofArea, 412.9019, 0.01);
+  EXPECT_EQ(numRoofSurfaces, 9);
 
   //m.save("intersect3.osm", true);
+  //openstudio::path outpath = resourcesPath() / toPath("model/Space_intersectSurfaces_degenerate3_after_intersect.osm");
+  //m.save(outpath, true);
+}
+
+/// Illustrates a fix for surface intersection getting stuck in a loop
+/// First of all we need to remove surfaces that overlap within the same space
+/// Second of all we use a different removeSpikes method that shrinks and expands the polygon
+TEST_F(ModelFixture, RemoveSpikesAndOverlaps) {
+
+  openstudio::path path = resourcesPath() / toPath("model/two_stories_pre_intersect.osm");
+  // openstudio::path path = resourcesPath() / toPath("model/whole_bldg_partially_matched.osm");
+  // openstudio::path path = resourcesPath() / toPath("model/7-7 Windows Complete.osm");
+  osversion::VersionTranslator vt;
+  boost::optional<model::Model> model = vt.loadModel(path);
+  ASSERT_TRUE(model) << "Failed to load " << path;
+
+  SpaceVector spaces = model->getModelObjects<Space>();
+  SpaceVector blacklist;
+
+  int nSurfaces = 0;
+  for (const auto& space : spaces) {
+    nSurfaces += space.surfaces().size();
+  }
+
+  for (const auto& space : spaces) {
+    std::string name = space.name().value();
+    Transformation spaceTransformation = space.transformation();
+
+    std::vector<Surface> surfaces = space.surfaces();
+    // What is this?
+    //if (name == "Space TZ46-3") {
+    //  int hello = 1;
+    //}
```

Comment:
This was throwing an unused-variable warning treated as error. I didn't know what to do with it